src/overlays/actors/ovl_En_Ru2/z_en_ru2.c

```diff
@@ -313,174 +394,227 @@ void func_80AF2BC0(EnRu2* this, PlayState* play) {
         if ((cue != NULL) && (cue->id == 3)) {
             Animation_Change(&this->skelAnime, animation, 1.0f, 0.0f, Animation_GetLastFrame(animation), ANIMMODE_ONCE,
                              0.0f);
-            this->action = 4;
+            this->action = ENRU2_RAISE_ARMS;
         }
     }
 }
 
-void func_80AF2C54(EnRu2* this, s32 arg1) {
-    if (arg1 != 0) {
-        this->action = 5;
+/**
+ * At the end of Ruto's arms-raising animation, cues the next action: spawning the
+ * Water Medallion.
+ */
+void EnRu2_HoldArmsUp(EnRu2* this, s32 doneRaising) {
+    if (doneRaising != 0) {
+        this->action = ENRU2_AWAIT_SPAWN_WATER_MEDALLION;
     }
 }
 
-void func_80AF2C68(EnRu2* this, PlayState* play) {
+/**
+ * Checks to see if the Water Medallion should spawn.
+ */
+void EnRu2_CheckWaterMedallion(EnRu2* this, PlayState* play) {
     CsCmdActorCue* cue;
 
     if (play->csCtx.state != CS_STATE_IDLE) {
         cue = play->csCtx.actorCues[6];
 
         if ((cue != NULL) && (cue->id == 2)) {
-            this->action = 6;
-            func_80AF2A38(this, play);
+            this->action = ENRU2_FINISH_WATER_MEDALLION_CS;
+            EnRu2_SpawnWaterMedallion(this, play);
         }
     }
 }
 
-void func_80AF2CB4(EnRu2* this, PlayState* play) {
-    func_80AF2AB4(this, play);
+/**
+ * Action 0
+ */
```

Comment:
Personally dont think we need these comments describing action numbers. This information can be gathered from the enum